quasselgrep/query.py

```diff
@@ -69,18 +69,32 @@ def __init__(self, cursor, options, text, timerange=None):
 			self.limit = 0
 
 		#TODO Consider changing this to equality for buffer
-		self.params = {
-			'text' : Param('text', 'backlog.message LIKE %(param)s'),
-			'user' : Param('user', 'quasseluser.username = %(param)s'),
-			'network' : Param('network', 'network.networkname LIKE %(param)s'),
-			'buffer' : Param('buffer', 'buffer.buffername LIKE %(param)s'),
-			'sender' : Param('sender', '(sender.sender = %(param)s OR sender.sender LIKE %(param)s)', ['sender_pattern']),
-			'fromtime' : Param('fromtime', 'backlog.time > %(param)s'),
-			'totime' : Param('totime', 'backlog.time < %(param)s'),
-			# SQLite can't handle tuple parameters, and they're not from user
-			# input so just include them directly in the string
-			'msg_types' : TypesParam(self.msg_types),
-		}
+		if options.ignorecase:
```

Comment:
To reduce code repetition, could you only alter the text parameter inside the `if` and set the rest of the params outside? Cheers